matrix-api.c

```diff
@@ -113,6 +115,7 @@ typedef struct {
     JsonParser *json_parser;
     char *body;
     size_t body_len;
+    gchar *content_encoding;
```

Comment:
If I were writing this today, I'd add a comment here to say what `content_encoding` actually was, and how its lifecycle is managed.

I'd also put it next to `content_type`, since the two are closely related.

Reply:
Do you mean something like this:

gchar *content_encoding; // http header: Content-Encoding, if avaiable or NULL

Reply:
well, yes. Possibly with slightly more words and fewer spelling mistakes ;)